databroker/src/grpc/kuksa_val_v1/val.rs

```diff
@@ -475,6 +548,100 @@ impl proto::val_server::Val for broker::DataBroker {
     }
 }
 
+async fn validate_entry_update(broker: &AuthorizedAccess<'_, '_>, request: &EntryUpdate, id: i32) -> Result<Pair<i32, broker::EntryUpdate>, Status> {
```

Comment:
Since the streamed update api uses a lot of code from the set method I tried to separate some common things into separate methods (I did not want to redundantly copy 120 lines of code inside the loop mechanism).

I'm still not satisfied with the complexity from streamed_update method right now, but I still have some struggles with Rust and it's borrowing concept, making it hard for me to improve it even more. It also feels like a lot of complexity comes from all the matches and results?